src/com/amazon/ion/impl/lite/IonSystemLite.java

```diff
@@ -822,4 +685,190 @@ public boolean valueIsSharedSymbolTable(IonValue value)
+            if (text != null && token.getSid() != UNKNOWN_SYMBOL_ID)
+            {
+                token = newSymbolToken(text, UNKNOWN_SYMBOL_ID);
+            }
+            value.setFieldNameSymbol(token);
+            return true;
+        }
+
+        // Copies any annotations found on the reader's current position over to the provided `value`.
+        // Returns `true` if more than any annotations were found/copied. If no annotations were found/copied,
+        // returns `false`.
+        private boolean cloneAnnotationsIfAny(IonValueLite value) {
+            // `getTypeAnnotationSymbols` returns a freshly allocated array, so we can safely modify it.
+            SymbolToken[] annotations = reader.getTypeAnnotationSymbols();
+            if (annotations.length == 0) {
+                return false;
+            }
+
+            for (int i = 0; i < annotations.length; i++)
+            {
+                SymbolToken token = annotations[i];
+                String text = token.getText();
+                if (text != null && token.getSid() != UNKNOWN_SYMBOL_ID )
+                {
+                    annotations[i] = newSymbolToken(text, UNKNOWN_SYMBOL_ID);
+                }
+            }
+            value.setTypeAnnotationSymbols(annotations);
+            return true;
+        }
+
+        // Appends the provided value to the container at the top of the container stack.
+        // Callers must guarantee that the container stack is not empty before invoking this.
+        private void attachToParent(IonValueLite value) {
+            // Get a reference to the container at the top of the container stack.
+            IonContainerLite parent = this.containerStack.get(this.containerStack.size() - 1);
+            // If this is the first child value with its symbol-is-present flag set to `true`,
+            // then we also need to set the parent's symbol-is-present flag to true as well.
+            boolean childSymbolIsPresent = value._isSymbolPresent();
+            boolean parentSymbolIsPresent = parent._isSymbolPresent();
+            parent._isSymbolPresent(parentSymbolIsPresent | childSymbolIsPresent);
+            // Append the child value to the end of the container.
+            parent.add(value);
+        }
+
+        // Materializes the Ion value over which the provided `reader` is currently positioned.
+        // If the reader is not positioned over a value, returns `null`.
+        public IonValueLite load(IonReader reader) {
+            // Set `this.reader` for the duration of the load() process.
+            this.reader = reader;
+
+            // If a previous attempt to read Ion data failed (because of invalid syntax, for example), the ValueLoader's
+            // `containerStack` member field can be left with residual data. Clearing it at the outset of this method
+            // call allows the ValueLoader to be reused after such failures.
+            containerStack.clear();
```

Comment:
🗺️ In the current implementation we always construct a new `ValueLoader`, so the failure mode described by the comment will never happen. If we decided to start reusing `ValueLoader` instances as an optimization (possibly by creating a single one per thread), this safeguard would become important.